plugins/notifications-backend/src/service/db.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+
+import knex, { Knex } from 'knex';
+import merge from 'lodash/merge';
+
+// initDB
+// creates DB client and tables
+export async function initDB(dbConfig: Config): Promise<Knex<any, any>> {
+  // create db client
+  const knexConfig = merge(
+    {},
+    { connection: { database: 'backstage_plugin_notifications' } },
+    dbConfig.getOptional('plugin.notifications'),
+    dbConfig.getOptional('plugin.notifications.knexConfig'),
+    dbConfig.get(),
+    dbConfig.getOptional('knexConfig'),
+  );
+  const dbClient = knex(knexConfig);
+
+  // create tables
+  if (!(await dbClient.schema.hasTable('messages'))) {
+    await dbClient.schema.createTable('messages', table => {
```

Comment:
These are normally handled by separate `migrations` directory that is handled by the knex client `.migrate` functionality. See other plugins as a reference.

Reply:
done, thanks, I didn't know about this option